package.json

```diff
@@ -38,11 +38,11 @@
     "test:watch": "jest --watch"
   },
   "dependencies": {
-    "@metamask/snaps-sdk": "^4.2.0",
-    "@metamask/utils": "^8.4.0",
+    "@metamask/snaps-sdk": "^4.4.1",
```

Comment:
Why do we need to upgrade this one?

Reply:
Thanks for checking in on this!

`@metamask/snaps-sdk` is also undergoing the [same changes as this PR](https://github.com/MetaMask/snaps/pull/2445/files#diff-0fec96bbb8cd0770a0e63a84651b058df8823606f960b521b7a67b792e0fc646), which are due to be released in `^4.4.3`. 

That change needs to be reflected in this package because we need to replace `superstruct` in all _nested_ dependencies of core. So we need to update the dependencies in all core packages so that both of the following are satisfied:

- `@metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`
- `@metamask/keyring-api@^6.4.1 > @metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`

What we're trying to avoid is a conflicting resolution like this, because the `NodeNext` option doesn't work if we have the ESM-incompatible `superstruct` anywhere in the dependency tree:
- `@metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`
- `@metamask/keyring-api@^6.4.1 > @metamask/snaps-sdk@^4.2.0 > superstruct@^1.0.3`

To achieve this, I'm first bumping `@metamask/snaps-sdk` to the latest released version to check for regressions or breakage that need fixing. 

Next step is to bump to the unreleased version. I was planning to do this in a separate PR, but it looks like I might as well do it here, which is why I've switched this back to draft. I'll set this PR as ready to review once https://github.com/MetaMask/snaps/pull/2445/ is merged and released.

Hope this clarifies things!

Reply:
@ccharly Thanks for checking in on this!

`@metamask/snaps-sdk` is also undergoing the [same changes as this PR](https://github.com/MetaMask/snaps/pull/2445/files#diff-0fec96bbb8cd0770a0e63a84651b058df8823606f960b521b7a67b792e0fc646), which are due to be released in `^4.4.3`. 

That change needs to be reflected in this package because we need to replace `superstruct` in all _nested_ dependencies of core. So we need to update the dependencies in all core packages so that both of the following are satisfied:

- `@metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`
- `@metamask/keyring-api@^6.4.1 > @metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`

What we're trying to avoid is a conflicting resolution like the following, because the `NodeNext` option doesn't work if we have the ESM-incompatible `superstruct` anywhere in the dependency tree:
- `@metamask/snaps-sdk@^4.4.3 > @metamask/superstruct@^3.0.0`
- `@metamask/keyring-api@^6.4.1 > @metamask/snaps-sdk@^4.2.0 > superstruct@^1.0.3`

To achieve this, I'm first bumping `@metamask/snaps-sdk` to the latest released version to check for regressions or breakage that need fixing. 

Next step is to bump to the unreleased version. I was planning to do this in a separate PR, but it looks like I might as well do it here, which is why I've switched this back to draft. I'll set this PR as ready to review once https://github.com/MetaMask/snaps/pull/2445/ is merged and released.

Hope this clarifies things!